src/Features/Core/Portable/MetadataAsSource/AbstractMetadataAsSourceService.DocCommentFormatter.cs

```diff
@@ -157,60 +137,73 @@ internal static ImmutableArray<string> Format(IDocumentationCommentFormattingSer
                 while (formattedCommentLinesBuilder is [.., { Length: 0 }])
                     formattedCommentLinesBuilder.RemoveAt(formattedCommentLinesBuilder.Count - 1);
 
-                return formattedCommentLinesBuilder.ToImmutableAndFree();
+                return formattedCommentLinesBuilder.ToImmutable();
             }
 
-            private static ImmutableArray<string> CreateWrappedTextFromRawText(string rawText)
+            private static void AddWrappedTextFromRawText(
+                string rawText, ArrayBuilder<string> result, string prefix = null)
             {
-                using var _ = ArrayBuilder<string>.GetInstance(out var lines);
+                if (string.IsNullOrWhiteSpace(rawText))
+                    return;
+
+                if (prefix != null)
+                {
+                    if (result.Count > 0)
+                        result.Add(string.Empty);
+
+                    result.Add(prefix);
+                }
 
                 // First split the string into constituent lines.
                 var split = rawText.Split(new[] { "\r\n" }, System.StringSplitOptions.None);
```

Comment:
would like a Span/Memory way to avoid this alloc as well.  But we're on netstandard2.